src/main/kotlin/com/cognifide/gradle/aem/pkg/PackagePlugin.kt

```diff
@@ -94,16 +98,19 @@ class PackagePlugin : CommonDefaultPlugin() {
         }
 
         register<PackageDelete>(PackageDelete.NAME) {
-            dependsOn(compose)
+            dependsOn(compose, validate)
```

Comment:
I am seeing this as ... you should even not to try upload then install (deploy) package if validation will tell us that package is broken. the intention is to have the most immediate feedback

Reply:
these tasks are requiring a built package because right now there is freedom in case of a way of specifying package coordinates (group, name, version) in file META-INF/vault/properties.xml.

in some earlier versions of GAP, there was no such requirement, and these coordinates were determined by convention, but the side effect of that was that we got more close relationships between tasks. After uploading a package to AEM, now GAP is in 100% sure that coordinates used to determine state of the package just uploaded will be correct after storing the package on instance. E.g AEM/CRX package manager has some logic related with "path" property specified in "properties.xml" and it is not sure that package will be stored on path specified. The method used is just mostly battle-tested and plays well with task agnostic API e.g snippets like `aem.sync { packageManager.deploy(file) } ` so that tasks and that API could use same "core" methods.